src/Interfaces/AbstractZonotope.jl

```diff
@@ -525,3 +525,52 @@ function constraints_list(Z::AbstractZonotope{N}; check_full_rank::Bool=true
     end
     return constraints
 end
+
+"""
+    rectify(Z::AbstractZonotope{N}) where {N}
+
+Concrete rectification of a zonotopic set.
+
+### Input
+
+- `Z` -- zonotope
+
+### Output
+
+The zonotope obtained by rectifying `Z`.
+
+### Algorithm
+
+This function implements [Theroem 3.1, 1].
+
+[1] *Singh, G., Gehr, T., Mirman, M., Püschel, M., & Vechev, M. (2018). Fast
+and effective robustness certification. In Advances in Neural Information
+Processing Systems (pp. 10802-10813).*
+"""
+function rectify!(Z::AbstractZonotope{N}) where {N}
+    c = Z.center
+    G = Z.generators
```

Comment:
An `AbstractZonotope` generally does not have fields `center` and `generators`. You have to restrict yourself to `Zonotope` or convert to that type first.